lib/index.d.ts

```diff
@@ -39,3 +45,76 @@ export class StateController<State> {
   /** @internal Unsubscribe the listener callback that was passed to subscribeUpdates */
   unsubscribeUpdates(listener: (props: Partial<State>) => void): void;
 }
+
+export class Component extends WebComponent {
+    /**
+     * Defines the state of the component, including all the properties required for rendering.
+     */
+    state: object;
+    /**
+     * Defines standard component configuration.
+     */
+    config: Component.ComponentConfigOptions;
+    /**
+     * Template helper functions defined in config object, and exposed to template code as $helpers.
+     * This getter uses the component's internal config cache.
+     */
+    helpers: object;
+    /**
+     * For use inside view templates, to create a child Panel component nested under this
+     * component, which will share its state object and update cycle.
+     */
+    child(tagName: string, config?: object): VNode;
+    /**
+     * Searches the component's Panel ancestors for the first component of the
+     * given type (HTML tag name).
+     */
+    findPanelParentByTagName(tagName: string): Component;
+    /**
+     * Fetches a value from the component's configuration map (a combination of
+     * values supplied in the config() getter and defaults applied automatically).
+     */
+    getConfig(key: string): any;
+    /**
+     * Executes the route handler matching the given URL fragment, and updates
+     * the URL, as though the user had navigated explicitly to that address.
+     */
+    navigate(fragment: string, stateUpdate?: object): void;
+    /**
+     * Sets a value in the component's configuration map after element
+     * initialization.
+     */
+    setConfig(key: string, val: any): void;
```

Comment:
setConfig can also use indexed types too. Basically let's stay away from any as much as we can. Panel is base library and strong typechecking here offers us the most benefits.